rlclientlib/logger/async_batcher.h

```diff
@@ -52,6 +65,7 @@ namespace reinforcement_learning { namespace logger {
     event_queue<TEvent> _queue;       // A queue to accumulate batch of events.
     size_t _send_high_water_mark;
     error_callback_fn* _perror_cb;
+    typename TSerializer<TEvent>::shared_state_t& _shared_state;
```

Comment:
who is going to own this guy? Looks like with current dummy version we end up having reference to non-existing stack variable - we are not using it, but it seems to be UB. If async_batcher is going to be owner, do we need it as input argument to ctr? Or am I missing something?

Reply:
In the full PR this is owned by live_model_impl and passed down the pipe.
I've chosen to make it unconditional because the template magic required to make this optional turned out to be way too complicated to be worth it. This UB goes away later when I introduce the class that holds the state.